packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2645,6 +2667,23 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         }
     }
 
+    private rollback(
+        type: ContainerMessageType,
+        content: any,
+        localOpMetadata: unknown,
+        opMetadata: Record<string, unknown> | undefined,
+    ) {
+        switch (type) {
+            case ContainerMessageType.FluidDataStoreOp:
+                // For Operations, call resubmitDataStoreOp which will find the right store
+                // and trigger resubmission on it.
+                this.dataStores.rollbackDataStoreOp(content, localOpMetadata);
+                break;
+            default:
+                unreachableCase(type as never, `Unknown ContainerMessageType: ${type}`);
```

Comment:
The `as never` shouldn't be required, the point of `unreachableCase` is to raise a compiler error if the switch statement doesn't cover all cases in the type.

Reply:
In other words, if `type` isn't narrowed to type `never` already, then it's not an unreachable case.